client/network/src/protocol/sync.rs

```diff
@@ -696,6 +699,10 @@ impl<B: BlockT> ChainSync<B> {
 						}
 						PeerSyncState::DownloadingStale(_) => {
 							peer.state = PeerSyncState::Available;
+							if blocks.is_empty() {
```

Comment:
Why are we not doing this for the other states as well?

Reply:
When doing full sync it is not that critical, as we'll get new blocks from other peers eventually.